### PR TITLE
[log-shipper] Add keepDeletedFilesOpenedFor option

### PR DESCRIPTION
**Pull request by nabokihms**
## Description
Add new option `keepDeletedFilesOpenedFor` to PodLoggingConfig and ClusterLoggingConfig.kubernetesPods.

The format is a string containing the time unit in hours and minutes: 30m, 1h, 2h30m, 24h.

## Why do we need it, and what problem does it solve?
Specifies the time to keep deleted files opened for reading. Vector will keep pods metadata for this time as well to read logs from deleted pods.

This option is useful in cases of a log storage unavailability or a network partition. Vector will keep log files opened until finally sending them to the destination. However, the following option may affect the resource consumption of the Vector and also flood a disk with deleted logs. Use it with caution.

Closes https://github.com/deckhouse/deckhouse/issues/10184

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [x] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: log-shipper
type: feature
summary:  Add keepDeletedFilesOpenedFor option. Now it is possible to configure how long Vector keeps deleted file in case of unavailability of a log storage (when Vector cannot send logs to a storage and the internal buffer is already filled). Before the change, Vector used to hold log files opened indefinitely, which can cause a node outage by flooding the disk space. The option makes this behaviour configurable.
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Lada7878. I've started the workflow for label 'deploy/web/test'...

File: modules/460-log-shipper/crds/doc-ru-cluster-logging-config.yaml [hunk 1]
```diff
@@ -21,6 +21,15 @@ spec:
                   description: |
                     Описывает правило сбора логов из подов кластера.
                   properties:
+                    keepDeletedFilesOpenedFor:
+                      description: |
+                        Указывает время, в течение которого удаленные файлы будут оставаться открытыми для чтения.
+                        Vector будет сохранять метаданные подов в течение этого времени, чтобы читать логи из удаленных подов.
+                        Эта опция полезна в случаях недоступности хранилища логов или сетевого разделения. Vector будет держать файлы логов открытыми до их окончательной отправки в пункт назначения.
+
+                        Однако применение опции может повлиять на потребление ресурсов Vector и также быть причиной нехватки места на диске из-за удаленных файлов с логами. Используйте с осторожностью.
```
Comment (z9r5) [inline modules/460-log-shipper/crds/doc-ru-cluster-logging-config.yaml:30]:
```suggestion
                        Применение опции может повлиять на потребление ресурсов Vector и также быть причиной нехватки места на диске из-за удаленных файлов с логами. Используйте с осторожностью.
```

File: modules/460-log-shipper/crds/cluster-logging-config.yaml [hunk 2]
```diff
@@ -59,6 +59,19 @@ spec:
...
                     Describes a rule for collecting logs from the cluster's pods.
                   properties:
+                    keepDeletedFilesOpenedFor:
+                      type: string
+                      description: |
+                        Specifies the time to keep deleted files opened for reading.
+                        Vector will keep pods metadata for this time as well to read logs from deleted pods.
+                        This option is useful in cases of a log storage unavailability or a network partition. Vector will keep log files opened until finally sending them to the destination.
+
+                        However, enabling this option may affect the resource consumption of the Vector and also flood a disk with deleted logs. Use it with caution.
```
Comment (z9r5) [inline modules/460-log-shipper/crds/cluster-logging-config.yaml:69]:
```suggestion
                        Enabling this option may affect the resource consumption of the Vector and also flood a disk with deleted logs. Use it with caution.
```

File: modules/460-log-shipper/crds/doc-ru-pod-logging-config.yaml [hunk 3]
```diff
@@ -4,12 +4,21 @@ spec:
...
           properties:
             spec:
               properties:
+                keepDeletedFilesOpenedFor:
+                  description: |
+                    Указывает время, в течение которого удаленные файлы будут оставаться открытыми для чтения.
+                    Vector будет сохранять метаданные подов в течение этого времени, чтобы читать логи из удаленных подов.
+                    Эта опция полезна в случаях недоступности хранилища логов или сетевого разделения. Vector будет держать файлы логов открытыми до их окончательной отправки в пункт назначения.
+
+                    Однако применение опции может повлиять на потребление ресурсов Vector и также быть причиной нехватки места на диске из-за удаленных файлов с логами. Используйте с осторожностью.
```
Comment (z9r5) [inline modules/460-log-shipper/crds/doc-ru-pod-logging-config.yaml:19]:
```suggestion
                    Применение опции может повлиять на потребление ресурсов Vector и также быть причиной нехватки места на диске из-за удаленных файлов с логами. Используйте с осторожностью.
```

File: modules/460-log-shipper/crds/pod-logging-config.yaml [hunk 4]
```diff
@@ -32,6 +32,16 @@ spec:
...
                 - clusterDestinationRefs
               properties:
+                keepDeletedFilesOpenedFor:
+                  type: string
+                  description: |
+                    Specifies the time to keep deleted files opened for reading.
+                    Vector will keep pods metadata for this time as well to read logs from deleted pods.
+                    This option is useful in cases of a log storage unavailability or a network partition. Vector will keep log files opened until finally sending them to the destination.
+
+                    However, enabling this option may affect the resource consumption of the Vector and also flood a disk with deleted logs. Use it with caution.
```
Comment (z9r5) [inline modules/460-log-shipper/crds/pod-logging-config.yaml:42]:
```suggestion
                    Enabling this option may affect the resource consumption of the Vector and also flood a disk with deleted logs. Use it with caution.
```

**Review by Lada7878 — APPROVED** [review]

**Review by z9r5 — APPROVED** [review]

File: docs/documentation/_includes/module-settings.liquid [hunk 5]
```diff
@@ -8,7 +8,7 @@
 {%- endif %}
 
 {%- if page.lang == 'ru' %}
-Чтобы настроить модуль, используйте custom resource `ModuleConfig` с именем `{{ moduleKebabName }}` ([подробнее о настройке Deckhouse...](../../)).
+Чтобы настроить модуль, используйте кастомный ресурс `ModuleConfig` с именем `{{ moduleKebabName }}` ([подробнее о настройке Deckhouse...](../../)).
```
Comment (z9r5) [inline docs/documentation/_includes/module-settings.liquid:11]:
```suggestion
Чтобы настроить модуль, используйте ресурс `ModuleConfig` с именем `{{ moduleKebabName }}` ([подробнее о настройке Deckhouse...](../../)).
```

**Review by vladimirGuryanov — APPROVED** [review]

**Pull request merged**